x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline.tsx

```diff
@@ -0,0 +1,235 @@
+  Name,
+  UpdateTitle,
+  UpdateDescription,
+  SaveTimeline,
+} from '../properties/helpers';
+import { NOTES_PANEL_WIDTH } from '../properties/notes_size';
+import { TIMELINE_TITLE, DESCRIPTION } from '../properties/translations';
+import { useCreateTimelineButton } from '../properties/use_create_timeline';
+import * as i18n from './translations';
+
+interface SaveTimelineButtonProps {
+  timelineId: string;
+  showOverlay: boolean;
+  toolTip?: string;
+  toggleSaveTimeline: () => void;
+  onSaveTimeline: SaveTimeline;
+  updateTitle: UpdateTitle;
+  updateDescription: UpdateDescription;
+}
+
+interface TimelineTitleAndDescriptionProps {
+  timelineId: string;
+  toggleSaveTimeline: () => void;
+  onSaveTimeline: SaveTimeline;
+  updateTitle: UpdateTitle;
+  updateDescription: UpdateDescription;
+}
+
+const Wrapper = styled(EuiModalBody)`
+  .euiFormRow {
+    max-width: none;
+  }
+
+  .euiFormControlLayout {
+    max-width: none;
+  }
+
+  .euiFieldText {
+    max-width: none;
+  }
+`;
+
+Wrapper.displayName = 'Wrapper';
+
+export const TimelineTitleAndDescription = React.memo<TimelineTitleAndDescriptionProps>(
+  ({ timelineId, toggleSaveTimeline, onSaveTimeline, updateTitle, updateDescription }) => {
+    const timeline = useShallowEqualSelector((state) =>
+      timelineSelectors.selectTimeline(state, timelineId)
+    );
+
+    const { description, isSaving, savedObjectId, title, timelineType } = timeline;
+
+    const handleClick = useCallback(() => {
+      onSaveTimeline({
+        ...timeline,
+        id: timelineId,
+      });
+    }, [onSaveTimeline, timeline, timelineId]);
+
+    const { getButton } = useCreateTimelineButton({ timelineId, timelineType });
+
+    const discardTimelineButton = useMemo(
+      () =>
+        getButton({
+          title:
+            timelineType === TimelineType.template
+              ? i18n.DISCARD_TIMELINE_TEMPLATE
+              : i18n.DISCARD_TIMELINE,
+          outline: true,
+          iconType: '',
+          fill: false,
+        }),
+      [getButton, timelineType]
+    );
+
+    const modalHeader =
+      savedObjectId == null
+        ? timelineType === TimelineType.template
+          ? i18n.SAVE_TIMELINE_TEMPLATE
+          : i18n.SAVE_TIMELINE
+        : timelineType === TimelineType.template
+        ? i18n.NAME_TIMELINE_TEMPLATE
+        : i18n.NAME_TIMELINE;
+
+    const saveButtonTitle =
+      savedObjectId == null
+        ? timelineType === TimelineType.template
+          ? i18n.SAVE_TIMELINE_TEMPLATE
+          : i18n.SAVE_TIMELINE
+        : i18n.SAVE;
+
+    return (
+      <>
+        {isSaving && (
+          <EuiProgress size="s" color="primary" position="absolute" data-test-subj="progress-bar" />
+        )}
+        <EuiModalHeader data-test-subj="modal-header">{modalHeader}</EuiModalHeader>
+
+        <Wrapper>
+          <EuiFlexItem grow={true}>
+            <EuiFormRow label={TIMELINE_TITLE}>
+              <Name
+                timelineId={timelineId}
```

Comment:
Consider focusing the `Title` input when the modal opens to eliminate the need to manually click into the input before typing:

![focus](https://user-images.githubusercontent.com/4459398/97197612-78c84200-1773-11eb-8b98-f4c056ca8554.gif)

Also consider removing the `Title` tooltip shown in the gif above, since it's no longer required in the new design.